### PR TITLE
Test for passing plugin parameters with dashes in names

### PR DESCRIPTION
**Pull request by jan-kolarik**
Requires https://github.com/rpm-software-management/dnf/pull/1839.

### CODE REVIEW

File: dnf-behave-tests/dnf/plugins-core/enable-plugins-dashes.feature [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+@bz1980712
+Scenario: Existing plugin containing dashes in the name could be disabled
+   When I execute dnf with args "repoquery empty --disableplugin=config-manager --verbose"
+   Then the exit code is 0
+    And stdout does not contain "config-manager"
+    And stderr does not contain "No matches found for the following disable plugin patterns: config-manager"
+
+@bz1980712
+Scenario: Existing plugin containing dashes in the name could be enabled
+   When I execute dnf with args "--disableplugin=config-manager"
```
Comment (m-blaha) [inline dnf-behave-tests/dnf/plugins-core/enable-plugins-dashes.feature:16]: `--disableplugin` only affects the current run of dnf, disablement of plugin is not persistent. That said, this line doesn't do anything useful and does not affect the following `repoquery --enableplugin...`
To simulate enabling a disabled plugin, you coud use `--disableplugin='*' --enableplugin='config-manager`

Comment (jan-kolarik) [reply dnf-behave-tests/dnf/plugins-core/enable-plugins-dashes.feature]: Thank you Marek, how could I've missed that :slightly_smiling_face:

File: dnf-behave-tests/dnf/plugins-core/enable-plugins-dashes.feature [hunk 2]
```diff
@@ -0,0 +1,33 @@
+Feature: Tests for enabling plugins containing dashes in names
+
+Background: Enable plugins
+  Given I do not disable plugins
+    And I use repository "dnf-ci-fedora"
```
Comment (m-blaha) [inline dnf-behave-tests/dnf/plugins-core/enable-plugins-dashes.feature:5]: Here you should make sure, that the config-manager plugin is really available and enabled. Either by checking `dnf -v` output or by checking the exit code of `dnf config-manager --help` command.

File: dnf-behave-tests/dnf/plugins-core/enable-plugins-dashes.feature [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+@bz1980712
+Scenario: Existing plugin containing dashes in the name could be disabled
+   When I execute dnf with args "repoquery empty --disableplugin=config-manager --verbose"
+   Then the exit code is 0
+    And stdout does not contain "config-manager"
```
Comment (m-blaha) [inline dnf-behave-tests/dnf/plugins-core/enable-plugins-dashes.feature:11]: I would prefer more accurate check here. The exact match would require defined set of available plugins, but at least something like `And stdout does not contain "Loaded plugins: .*config-manager"` would be better. Or maybe `stdout matches line by line` step might be useful (at least for positive cases).

Comment (jan-kolarik) [reply dnf-behave-tests/dnf/plugins-core/enable-plugins-dashes.feature]: Definitely, this is better and I didn't check that `contain` step is using pattern matching.

File: dnf-behave-tests/dnf/plugins-core/enable-plugins-dashes.feature (hunk 2 above)
Comment (jan-kolarik) [reply dnf-behave-tests/dnf/plugins-core/enable-plugins-dashes.feature:5]: I agree, thanks.

**Pull request merged**